pkg/git/getter.go

```diff
@@ -3,39 +3,36 @@ package git
 
 import (
 	"fmt"
-
-	"github.com/hashicorp/go-getter"
 	"kcl-lang.io/kpm/pkg/constants"
 )
 
-var goGetterGetters = map[string]getter.Getter{
-	"git": new(getter.GitGetter),
-}
-
-var goGetterNoDetectors = []getter.Detector{}
-
 const GIT_PROTOCOL = "git::"
 
 func ForceProtocol(url, protocol string) string {
 	return protocol + url
 }
 
-// ForceGitUrl will add the branch, tag or commit to the git URL and force it to the git protocol
-// `<URL>` will return `Git::<URL>?ref=<branch|tag|commit>`
+// ForceGitUrl will add the subpackage and branch, tag or commit to the git URL and force it to the git protocol
+// `<URL>` will return `Git::<URL>//<SubPackage>?ref=<branch|tag|commit>`
 func (cloneOpts *CloneOptions) ForceGitUrl() (string, error) {
 	if err := cloneOpts.Validate(); err != nil {
 		return "", nil
 	}
 
+	newRepoUrl := cloneOpts.RepoURL
+	if cloneOpts.SubPackage != "" {
+		newRepoUrl += "//" + cloneOpts.SubPackage
+	}
```

Comment:
The implementation idea here is still wrong, the idea we mentioned before is **clone the entire repo. Recursively finds the dependency with the specified package name and loads it**

Reply:
Okay will open a PR with that changes